### PR TITLE
C/C++: Add vcpkg 0.0.113 (or latest) (Linux, Windows)

### PR DESCRIPTION
**Pull request by vmapetr**
This PR fixes https://github.com/Microsoft/azure-pipelines-image-generation/issues/242 Adds vcpkg 0.0.113 (or latest) (Linux, Windows):
* Implemented script to install vcpkg on Windows
* Implemented script to install vcpkg on Linux
* Implemented separate installation script for GNU Cpp Compiler (g++) which is required for vcpkg on Linux.

**For discussion**
* Should some of the vcpkg packages to be pre-installed as per one of the issue comments?

**Tests**
* Run scripts manually on local boxes (Windows 10, Ubuntu 16.04)
* Via Azure VM provisioning and then verifying vcpckg (`vcpkg install openssl`)

### CODE REVIEW

**Comment by sgrebnov** [comment]
@vijayma, @azureDaveOps, @madhurig, @vidStaheli - could you please review this PR

File: images/linux/scripts/installers/gcc.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+################################################################################
+
+# Source the helpers for use with the script
+source $HELPER_SCRIPTS/document.sh
+
+# Install GNU C++ compiler
+add-apt-repository ppa:ubuntu-toolchain-r/test -y
+apt-get update -y
+apt-get install g++-7 -y
+
```
Comment (DavidStaheli) [inline images/linux/scripts/installers/gcc.sh]: The build-essential package is already installed [here](https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/linux/scripts/installers/build-essential.sh) and includes gcc 5, so this installation of gcc 7 is a welcome improvement.

**Comment by mitza-oci** [comment]
How should a project using the pre-installed `vcpkg` discover the location of the `installed` directory for use in the later build steps?  We could search the path for `vcpkg` but it seems like this shouldn't be needed in each project.

Our current azure-pipelines.yml for reference: https://github.com/objectcomputing/OpenDDS/blob/38e578946bdfeb75c3ee266528decf934ff29a49/azure-pipelines.yml#L22

**Comment by sgrebnov** [comment]
@mitza-oci - Please confirm the ideal scenario for you to have [VCPKG_ROOT](https://github.com/Microsoft/vcpkg/blob/1d5e22919fcfeba3fe513248e73395c42ac18ae4/docs/users/config-environment.md#vcpkg_root) env variable to be pre-defined on VM?

PS.  Similar to appveyor (https://www.appveyor.com/docs/lang/cpp/#vc-packaging-tool) I think it is also safe to consider vcpkg is pre-installed to a specific folder ('C:\vcpkg' on Windows and '/usr/local/share/vcpkg' on Linux).

**Comment by mitza-oci** [comment]
Yes, sounds like that would work.

**Comment by ras0219-msft** [comment]
This will cause problems for users that are currently cloning their own vcpkg instance -- when they build their own vcpkg executable and run it, the `VCPKG_ROOT` environment variable will override their local paths with the central one in the image.

This is undesirable because the model of vcpkg is that the VCPKG_ROOT location stores the version information for what libraries to install. If a user has cloned their own copy, they presumably wish to install those version ;).

We would instead recommend that users either clone their own copy of vcpkg or hardcode the path for the azure pipelines images.

**Comment by max-lobanov** [comment]
Hi,

It looks like these changes have overridden the default `VCPKG_ROOT` vcpkg variable that should point to the `$HOME/.vcpkg`. It causes permission issues for our customers on Linux images
To fix the issue we need to rename `VCPKG_ROOT` variable to something like `VCPKG_INSTALLATION_ROOT`. So `VCPKG_INSTALLATION_ROOT` will point to the vcpkg installation dir and `VCPKG_ROOT` will point to the `$HOME/.vcpkg` by default
As a temporary workaround customers need to override `VCPKG_ROOT` to `$HOME/.vcpkg` in their scripts before using  `vcpkg`.

@DavidStaheli , I think we need to log bug for this issue and fix it. Please confirm

cc: @ras0219-msft, @sgrebnov

**Comment by DavidStaheli** [comment]
I created issue #769 to fix this.

**Pull request merged**